src/sync/index.ts

```diff
@@ -3,18 +3,90 @@
  */
 
 import {EventEmitter} from "events";
+import {IBeaconChain} from "../chain";
+import {INetwork} from "../network";
+import {OpPool} from "../opPool";
+import {IEth1Notifier} from "../eth1";
+import {IBeaconDb} from "../db";
+import {SyncRpc} from "./rpc";
+import {RegularSync} from "./regular";
+import {ReputationStore} from "./reputation";
+import {ILogger} from "../logger";
 
 /**
- * The Sync service handles everything related to receiving objects via p2p from other nodes
+ * The Sync service syncing data between the network and the local chain
  * The strategy may differ depending on whether the chain is synced or not
  */
 export class Sync extends EventEmitter {
-  private network;
+  private opts;
+  private chain: IBeaconChain;
+  private network: INetwork;
+  private opPool: OpPool;
+  private eth1: IEth1Notifier;
+  private db: IBeaconDb;
+  private rpc: SyncRpc;
+  private reps: ReputationStore;
+  private logger: ILogger;
+  private syncer;
 
-  public constructor(opts, {network}) {
+  public constructor(opts, {chain, db, eth1, network, opPool, reps, logger}) {
     super();
+    this.opts = opts;
+    this.chain = chain;
+    this.db = db;
+    this.eth1 = eth1;
     this.network = network;
+    this.opPool = opPool;
+    this.reps = reps;
+    this.logger = logger;
+    this.rpc = new SyncRpc(opts, {db, chain, network, reps, logger});
+  }
+
+  public async isSynced(): Promise<boolean> {
+    if (!await this.eth1.isAfterEth2Genesis()) {
+      return true;
+    }
+    try {
+      const bestSlot = await this.db.getChainHeadSlot();
+      const bestSlotByPeers = this.network.getPeers()
+        .map((peerInfo) => this.reps.get(peerInfo.id.toB58String()))
```

Comment:
something is telling me from a high-level we may need an `await` in here if we're querying from peers, hard to tell im not sure what reps is

Reply:
await isn't needed here. reps is a ReputationStore.

Reply:
mmm gothcya, so this function takes exisiting data, and syncs up current db. all good.

Reply:
Yeah, the ReputationStore tracks each peer's best slot (among other things). We're looping through here to find the best slot according to our peers.